pkg/acsengine/defaults.go

```diff
@@ -675,6 +668,19 @@ func setStorageDefaults(a *api.Properties) {
 	}
 }
 
+func setHostedMasterOverrides(a *api.Properties) {
+	a.HostedMasterProfile.Subnet = DefaultKubernetesMasterSubnet
+
+	for _, profile := range a.AgentPoolProfiles {
+		// don't default Distro for OpenShift
+		if !a.OrchestratorProfile.IsOpenShift() {
+			if profile.Distro == "" {
```

Comment:
since `setAgentNetworkDefaults` happens before this, profile.Distro won't be empty by the time we reach this function